### ISSUE TITLE
Feature request: EASGD

### ISSUE THREAD

**Issue by jinxin0924** [tensorflow/tensorflow#12472]
EASGD is a very useful algorithm for distributed asynchronous training, which based on an elastic force which links the parameters of workers with a center variable stored by the parameter server. This allows the local variables to fluctuate further from the center variable, which in theory allows for more exploration of the parameter space. 

In practice I find it usually outperforms ordinary ASGD.  I have been working on it  for some days and I can make a pull request if you are interested in it.

## The link address of the paper
[Deep learning with Elastic Averaging SGD](https://arxiv.org/pdf/1412.6651.pdf)

**Comment by ali01** [comment]
We currently don't have anybody working on this. It would be great if you could help us by working on this and submitting a PR. Let us know if you need further clarification. Thanks!

**Comment by jinxin0924** [comment]
Hi @ali01,
I need some suggestions. Since the local variables and global variables are always not same, do you think it is important or not to know the global training loss/accuracy related to global variables?

**Comment by tmulc18** [comment]
@jinxin0924 I'm currently working on implementing ADAG. I'm stuck because I don't know how to create local variables that aren't communicated with the parameter server.   Do you have any code the demos this?

**Comment by jinxin0924** [comment]
@tmulc18 
I use CustomGetter to create local variables and global variables at the same time. In addition, you can take a loot at [Model Average implementation](https://github.com/tensorflow/tensorflow/pull/11581/files), they create local variables in the optimizer.

**Comment by ashimb9** [comment]
Could this issue please be closed since the relevant changes appear to have been merged? Thanks!
